packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -180,127 +130,7 @@ ReferenceArrayInput.defaultProps = {
     sort: { field: 'id', order: 'DESC' },
 };
 
-const sanitizeRestProps = ({
-    filterToQuery,
-    perPage,
-    reference,
-    referenceSource,
-    resource,
-    ...rest
-}: any) => sanitizeInputRestProps(rest);
-
-export type ReferenceArrayInputViewProps = CommonInputProps & {
-    allowEmpty?: boolean;
-    children: ReactElement;
-    choices: any[];
-    className?: string;
-    error?: string;
-    id: string;
-    isFetching: boolean;
-    isLoading: boolean;
-    onChange: any;
-    options?: any;
-    reference: string;
-    resource?: string;
-    setFilter: (v: string) => void;
-    setPagination: (pagination: PaginationPayload) => void;
-    setSort: (sort: SortPayload) => void;
-    translate: Translate;
-    warning?: string;
-};
-
-export const ReferenceArrayInputView = ({
-    allowEmpty,
-    children,
-    choices,
-    className,
-    error,
-    field,
-    fieldState,
-    formState,
-    isFetching,
-    isLoading,
-    isRequired,
-    label,
-    onChange,
-    options,
-    reference,
-    resource,
-    setFilter,
-    setPagination,
-    setSort,
-    source,
-    translate,
-    warning,
-    ...rest
-}: ReferenceArrayInputViewProps) => {
-    const translatedLabel =
-        typeof label === 'string'
-            ? translate(
-                  ...getFieldLabelTranslationArgs({
-                      label,
-                      resource,
-                      source,
-                  })
-              )
-            : label;
-
-    if (error) {
-        return <ReferenceError label={translatedLabel} error={error} />;
```

Comment:
what happens in case of error?

Reply:
We may have to add side effects functions like everywhere else (`onSuccess`, `onError`)?